### ISSUE TITLE
Add curl commands to SSO booster interaction

### ISSUE THREAD

**Issue by rhoads-zach** [openshiftio/appdev-documentation#304]
Per conversation with @Ladicek, add interaction examples using raw `curl` commands in addition to using the provided script.

@Ladicek please provide any additional detail if i missed anything.

**Comment by Ladicek** [comment]
The provided application is able to output a bash script using `curl`: `java -jar target/sso-client.jar  --outputCurlScript demo.sh`. The resulting script (`demo.sh`) has like 50 lines, but it all boils down to 2 `curl` invocations: first to obtain the token, second to access the secured application using the token.

If we were able to demonstrate this, it would IMHO clarify what's really going on behind the scenes. Just using the provided application is a bit easier, but also more opaque.

**Comment by inoxx03** [comment]
Script output example for reference:

```
#!/bin/bash
# A script template for performing the query via cURL
if [ "$#" -lt 1 ]
then
     echo "Usage: $0 ENDPOINT_NAME"
     exit
fi

REALM=master
USER=alice
PASSWORD=password
CLIENT_ID=demoapp
SECRET=1daa57a2-b60e-468b-a3ac-25bd2dc2eadc
APP=$1

echo "Using APP=${APP}"

declare -a routes
routes=($(oc get routes))
count=${#routes[@]}

# Read in the oc get routes information to determine SSO_HOST and endpoint routes
while read -r line
do
    #echo "$line"
    routeInfo=($line[@])
    #echo "  + ${routeInfo[0]}"
    if [ "${routeInfo[0]}" == "secure-sso" ]
    then
      SSO_HOST="https://${routeInfo[1]}"
      echo "SSH_HOST=${SSO_HOST}"
    fi
    if [ "${routeInfo[0]}" == "${APP}" ]
    then
      APP_URL="http://${routeInfo[1]}"
      echo "using APP_URL=${APP_URL}"
    fi
done < <(oc get routes)

GET="curl -sk -X POST $SSO_HOST/auth/realms/$REALM/protocol/openid-connect/token -d grant_type=password -d username=$USER -d client_secret=$SECRET -d password=$PASSWORD -d client_id=$CLIENT_ID"
#GET="wget --no-check-certificate -O - --post-data=grant_type=password&username=${USER}&password=${PASSWORD}&client_id=${CLIENT_ID}&client_secret=${SECRET} ${SSO_HOST}/auth/realms/${REALM}/protocol/openid-connect/token"
#echo ">>> HTTP Token query"
echo $GET

auth_result=$($GET)
access_token=$(echo -e "$auth_result" | awk -F"," '{print $1}' | awk -F":" '{print $2}' | sed s/\"//g | tr -d ' ')

echo "<<< TOKEN Received"
echo $access_token

echo ">>> Greeting"
curl -kv $APP_URL/greeting -H "Authorization:Bearer $access_token"
```

**Comment by inoxx03** [comment]
@Ladicek 
Would the curl commands in question be the follwing:

* ```curl -sk -X POST $SSO_HOST/auth/realms/$REALM/protocol/openid-connect/token -d grant_type=password -d username=$USER -d client_secret=$SECRET -d password=$PASSWORD -d client_id=$CLIENT_ID```

* ```curl -kv $APP_URL/greeting -H "Authorization:Bearer $access_token"```

**Comment by rhoads-zach** [comment]
@inoxx03 take a look at #332 and work with @lavernw1 on the right path forward both of these issues.

**Comment by Ladicek** [comment]
@inoxx03 That's exactly it.

**Comment by inoxx03** [comment]
@Ladicek For some reason unknown to me, when I interact with the jarfile to get a response from the ```/greeting``` I get the expected "Hello World" response form the ```/greeting``` endpoint. When I use the commands from the script, however, I am still able to issue the request, but the respons comes back as ```401``` Unauthorized.
I'm pretty sure I'm missing something really basic here, please advise.

**Comment by Ladicek** [comment]
OK, so here's what I'm doing:

```bash
minishift start # with the CentOS image etc.
# OpenShift gets started on 192.168.99.100

cd .../wfswarm-rest-http-secured/
# using the Swarm variant here, but other SSO boosters are similar

cd sso
mvn clean fabric8:deploy
oc get route secure-sso # remember the URL for later use

cd ../app/
mvn clean fabric8:deploy -Popenshift -DskipTests -DSSO_AUTH_SERVER_URL=https://secure-sso-myproject.192.168.99.100.nip.io/auth

MY_TOKEN=$(curl -s -k -X POST https://secure-sso-myproject.192.168.99.100.nip.io/auth/realms/master/protocol/openid-connect/token -d client_id=demoapp -d client_secret=1daa57a2-b60e-468b-a3ac-25bd2dc2eadc -d grant_type=password -d username=alice -d password=password | jq -r .access_token)
# notice the `jq` invocation, which extracts the token value from the JSON response

oc get route wfswarm-rest-http-secured # remember the URL
curl -k -v http://wfswarm-rest-http-secured-myproject.192.168.99.100.nip.io/greeting -H "Authorization: Bearer $MY_TOKEN"
```

**Comment by inoxx03** [comment]
Resolved in PR #382
